### PR TITLE
refactor: [phase1] remove eth-types

### PR DESCRIPTION
**Pull request by lightsing**
revm repo needs update after this merge.

### CODE REVIEW

File: crates/primitives/src/types/tx.rs [hunk 1]
```diff
@@ -0,0 +1,550 @@
...
+        let payload_length = self.fields_len();
+        1 + Header {
+            list: true,
+            payload_length,
+        }
+        .length()
+            + payload_length
+    }
+
+    fn encode_2718(&self, out: &mut dyn BufMut) {
```
Comment (lispc) [inline crates/primitives/src/types/tx.rs:406]: => encode_eip2718.    Always prefix with "eip"

File: crates/primitives/src/lib.rs [hunk 2]
```diff
@@ -176,64 +139,28 @@ pub trait BlockChunkExt: BlockTraceExt + BlockTraceRevmExt {
...
-/// Revm extension trait for Transaction
-pub trait Transaction {
-    /// get the raw tx type
-    fn raw_type(&self) -> u8;
-    /// check if the tx is l1 tx
-    fn is_l1_tx(&self) -> bool {
-        self.raw_type() == 0x7e
+    /// execution_results
+    fn execution_results(&self, _tx_id: usize) -> Option<&eth_types::l2_types::ExecutionResult> {
+        None
```
Comment (lispc) [inline crates/primitives/src/lib.rs:145]: remove this?  nowadays the sbv is quite stable, we have little need for the "post_check"?

Comment (lightsing) [reply crates/primitives/src/lib.rs:145]: removed

**Comment by lispc** [comment]
no need to add both {v1,v2}_bytecode.{bin,txt} all?  txt and bin are same thing actually?

File: crates/primitives/src/types/mod.rs [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+    flatten_proofs: Vec<(B256, Bytes)>,
+}
+
+/// Legacy format of block trace
+#[derive(
+    rkyv::Archive, rkyv::Serialize, rkyv::Deserialize, Serialize, Deserialize, Default, Debug, Clone,
+)]
+#[archive(check_bytes)]
+#[archive_attr(derive(Debug, Hash, PartialEq, Eq))]
+pub struct BlockTrace {
```
Comment (lispc) [inline crates/primitives/src/types/mod.rs:92]: add a ref link to https://github.com/scroll-tech/go-ethereum/blob/develop/core/types/l2trace.go

File: crates/primitives/src/types/mod.rs [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+    /// chain id
+    #[serde(rename = "chainID", default)]
+    pub chain_id: u64,
+    /// coinbase's status AFTER execution
+    pub coinbase: Coinbase,
```
Comment (lispc) [inline crates/primitives/src/types/mod.rs:97]: change this comment.  it is actually just coinbase address

File: crates/primitives/src/types/mod.rs [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+    /// block
+    pub header: BlockHeader,
+    /// txs
+    pub transactions: Vec<TransactionTrace>,
+    /// Accessed bytecodes with hashes
+    #[serde(default)]
```
Comment (lispc) [inline crates/primitives/src/types/mod.rs:103]: i want to reject empty bytecodes field.  All these are ancient invalid traces.

**Review by lispc — APPROVED** [review]

**Comment by lightsing** [comment]
> 1. no need to add both {v1,v2}_bytecode.{bin,txt} all?  txt and bin are same thing actually?
> 2. need to update README related to '--disable-checks'?

add both file is for others can verify that they are same thing, as it's originally hex encoded.

**Review by lispc — APPROVED** [review]

**Pull request merged**